### ISSUE TITLE
discord::Connection::recv_event stops producing events after some time

### ISSUE THREAD

**Issue by saethlin** [tinaun/discord-rs#15]
I don't get any `Err` from `recv_event()`, just... nothing after a while.

**Comment by saethlin** [comment]
@tinaun https://github.com/tinaun/discord-rs/blob/50c633d38e4780d83bcd1974784ebb7762608023/src/connection.rs#L71 looks like it might be the culprit? (this bug is rather annoying, I'm gonna see if I can manage a fix)

**Comment by tinaun** [comment]
yeah, it doesn't handle reconnects atm. -- i think it should quit entirely when it does, at least for now.

anyway, sorry for ignoring this crate for a bit - now that tokio and futures 0.2 seem to be on the horizion im going to see if i can update to that

**Comment by saethlin** [comment]
Would I need to use Tokio to contribute a reconnect implementation?

**Comment by tinaun** [comment]
not directly, at least. what i would do would be to have the inner websocket connection shutdown completely, and then create a new connection with a reconnect event as described in discords docs.
